

# 2. Beagle file

## 2.1 run angsd

```bash
#!/bin/bash -l
#SBATCH -A naiss2023-5-222
#SBATCH -p node
#SBATCH -n 1
#SBATCH -C mem256GB
#SBATCH --array=17,19,20,28,2,3,41
#SBATCH -t 1-00:00:0
#SBATCH -J beagle
#SBATCH -e beagle_%A_%a.err
#SBATCH -o begle_%A_%a.out
#SBATCH --mail-type=all
#SBATCH --mail-user=qiaoling.deng@imbim.uu.se

#Load modules
module load bioinfo-tools
module load ANGSD/0.933
#module load python/2.7.15


######################################################################################
# CONDUCT GENOTYPE LIKELIHOOD ESTIMATION USING ANGSD v.1.11
# Will output the following files per chromosome:
#  Beagle (genotype likelihood format)


# Qiaoling Deng, 2024/1/25
######################################################################################

#STEP 1: Specify out directory
IN_DIR=/proj/snic2020-2-19/private/cisco/qiaoling/Bam_Files/Final_Merged_Bam
OUT_DIR=/proj/snic2020-2-19/private/cisco/qiaoling/Bam_Files/ANGSD_analysis/Beagle_file
LIST_DIR=/proj/snic2020-2-19/private/cisco/qiaoling/Bam_Files/ANGSD_analysis/List

#STEP 2: Define paths to Refference genome
REFGENOME=/proj/snic2020-2-19/private/cisco/qiaoling/new_assembly/European_cisco_w_extra_data.ipa1.8.0.purged.primary.fa
REF_INDEXED=/proj/snic2020-2-19/private/cisco/qiaoling/new_assembly/European_cisco_w_extra_data.ipa1.8.0.purged.primary.fa.fai

#STEP 3: #Specify chunk list
CHUNK_LIST=/proj/snic2020-2-19/private/cisco/qiaoling/new_assembly/CHR_chunk/Cisco_Chunk.list
CHUNK_NAMES=$(cat $CHUNK_LIST | sed -n ${SLURM_ARRAY_TASK_ID}p)
CHUNK_ID=${CHUNK_NAMES/.list}
CHUNK_DIR=/proj/snic2020-2-19/private/cisco/qiaoling/new_assembly/CHR_chunk

###########################################################################################

cd $IN_DIR

#STEP 4: Run ANGSD
angsd -bam $LIST_DIR/list_cisco_all_bam.list \
-ref $REFGENOME -fai $REF_INDEXED \
-rf $CHUNK_DIR/$CHUNK_NAMES \
-uniqueOnly 1 -remove_bads 1 -only_proper_pairs 1 -minMapQ 20 -minQ 20 \
-GL 2 -trim 0 -doMajorMinor 4 -doMaf 1 -doPost 2 -doGlf 2 -minMaf 0.1 -SNP_pval 1e-6  -docounts 1 \
-out $OUT_DIR/Cisco_aassembly2_all_bam_MinMAF_01_${CHUNK_ID} \
-nThreads 8


```

## 2.2 Site.txt

```bash

cd /proj/snic2020-2-19/private/cisco/qiaoling/Bam_Files/ANGSD_analysis/Beagle_file

#for f in Cisco_aassembly2_all_bam_MinMAF_01_chr_chunk_*.beagle.gz ; do
#    zcat $f | tail -n +2 | gzip >> $f.new
#done


#for f in Cisco_aassembly2_all_bam_MinMAF_01_chr_chunk*gz.new ; do
 #   cat $f >> Cisco_ANGSD_MAF0.1_Filtered_AllSamples_Output_nochr.beagle.gz
#done

zcat Cisco_ANGSD_MAF0.1_Filtered_AllSamples_Output_nochr.beagle.gz | cut -f1 | rev | sed 's/_/\t/' | rev > ANGSD_Cisco_aassembly2_MAF_01_sites.txt
#######check 
head -n 10  ANGSD_Cisco_aassembly2_MAF_01_sites.txt

########index sites
module load bioinfo-tools
module load ANGSD/0.933
angsd sites index  ANGSD_Cisco_aassembly2_MAF_01_sites.txt
```

