# 5.Gene_list

## 5.1 make a HDR bed file for each genome-wide screen--R

```R
#step1: make bed file for high differentiation gene list
###------------------------------------ 
setwd("D:/1_Cisco_remapping/7_remapping/2_ANGSD/2_GWAS")
load("gwas3_Kalix_with_mt.Rdata")
head(gwas3_Kalix_with_mt)
gwas3_Kalix_with_mt$logp <- -log10(gwas3_Kalix_with_mt$pvalue)

cutoff_0.001percent_thre <- quantile(gwas3_Kalix_with_mt$pvalue, 0.00001)
cutoff_0.01percent_thre <- quantile(gwas3_Kalix_with_mt$pvalue, 0.0001)

#threshold
gwas3_HDR_threshold <- gwas3_Kalix_with_mt[gwas3_Kalix_with_mt$pvalue < cutoff_0.01percent_thre,]
gwas3_HDR_threshold$log10_pvalue <- -log10(gwas3_HDR_threshold$pvalue)
head(gwas3_HDR_threshold)
gwas3_threshold <- gwas3_HDR_threshold[, c(1, 2, 2, 3, 4, 5, 11)]
head(gwas3_threshold)
gwas3_threshold[which.max(gwas3_threshold$log10_pvalue), ]
# Chromosome Position Position.1 Major Minor Frequency log10_pvalue
write.table(gwas3_threshold, "gwas3_threshold.bed",sep = "\t", col.names = FALSE,row.names = FALSE, quote = FALSE)

```

## 5.2 range_100kb--- linux

```bash
##range_SNP_list in linux
module load bioinfo-tools
module load BEDTools/2.29.2
module load python3
#Copy the original second column and put it in the last column 
awk -v OFS='\t' '{print $1, $2, $3, $4, $5, $6, $7,$2}' gwas3_threshold.bed > gwas3_threshold_inter1.bed
######Expand columns 2 and 3 by 100kb and remove carriage returns
sed 's/\r//g'  gwas3_threshold_inter1.bed > gwas3_threshold_inter2.bed
awk '{$2 = ($2 - 100000 < 0) ? 0 : $2 = $2 - 100000; $3 = $3 + 100000; print }' OFS='\t' gwas3_threshold_inter2.bed > gwas3_threshold_range100kb.bed
```

## 5.3 run py code

```bash
#3 run py script for gene list
# Run the gff file and bed file separately for chromosomes
GENELIST_PY=/proj/snic2020-2-19/private/cisco/qiaoling/Bam_Files/Annotation/Signal_list/1_gwas1/linked.py

#bed file | each chr
chromosomes=($(cut -f1 gwas3_threshold_range100kb.bed  | sort | uniq))
for chromosome in "${chromosomes[@]}"; do filename="gwas3_range_${chromosome}.bed"; grep "^$chromosome\s" gwas3_threshold_range100kb.bed > ./1_each_chr_gwas3/"$filename"; done


#### gff file | each chr
chrs=($(cut -f1 $GFF  | sort | uniq))
for chr in "${chrs[@]}"; do filename="gff_${chr}.gff"; grep "^$chr\s" $GFF > 1_gff_each_chr/"$filename"; done

#my gff_each_chr save atï¼š
GFF_each=/proj/snic2020-2-19/private/cisco/qiaoling/Bam_Files/Annotation/Signal_list/1_brackish/1_gff_each_chr
INPUT=/proj/snic2020-2-19/private/cisco/qiaoling/Bam_Files/Annotation/Signal_list/0_gene_list/3_kalix/1_each_chr_gwas3
OUTPUT=/proj/snic2020-2-19/private/cisco/qiaoling/Bam_Files/Annotation/Signal_list/0_gene_list/3_kalix/2_output
#run py script
GENELIST_PY=/proj/snic2020-2-19/private/cisco/qiaoling/Bam_Files/Annotation/Signal_list/1_gwas1/linked.py
for chromosome in "${chromosomes[@]}"; do 
python3 $GENELIST_PY  -g $GFF_each/"gff_${chromosome}.gff" -i "gwas3_range_${chromosome}.bed" -o output_gwas3_${chromosome};done

cat output_gwas* >> output_gwas3.bed

#check exon_variant 
awk '$9=="exon_variant" {print}' output_gwas3*.bed

cp -a output_gwas3.bed /home/qiade993/dql/Cisco/New_assembly/
```

```bash
###overlap gene
cd /proj/snic2020-2-19/private/cisco/qiaoling/Bam_Files/Annotation/Signal_list/0_gene_list/6_overlap
#SNP overlap

cut -f8 output_gwas4.bed |sort |uniq > gwas4_HDR_SNP.txt
cut -f8 output_gwas3.bed |sort |uniq > gwas3_HDR_SNP.txt

##gene_overlap
comm -12 <(sort gwas3_HDR_SNP.txt) <(sort gwas4_HDR_SNP.txt) > gwas3_gwas4_SNP_overlap.txt

########################
cd /proj/snic2020-2-19/private/cisco/qiaoling/Bam_Files/Annotation/Signal_list/0_gene_list/6_overlap

awk 'BEGIN {OFS="\t"}  {print $1, $8, $8, $4, $5, $6, $7, $9, $10, $11}' output_gwas3_sort.bed > try_gwas3.bed
awk 'BEGIN {OFS="\t"}  {print $1, $8, $8, $4, $5, $6, $7, $9, $10, $11}' output_gwas4_sort.bed > try_gwas4.bed
bedtools intersect -a try_gwas3.bed -b try_gwas4.bed -wb> gwas3_gwas4_overlap_SNP.bed

awk '{print $9}' gwas3_gwas4_overlap_SNP.bed | sort | uniq | wc -l
awk '{print $9}' gwas3_gwas4_overlap_SNP.bed | sort | uniq > gwas3_gwas4_overlap_Gene.list

```

### ==linked.py==  #python script 

```python
#! /usr/bin/python
import argparse
import re
import os

def parse_gff(gff):
    gene_info_dict = {}
    lines = gff.readlines()
    for line  in lines:
        line = line.strip()
        info = line.split()
        if re.search('ID=',info[8]):
            id = info[8].split(';')[0].split('=')[1]
        if re.search('Parent=',info[8]):
            parent = info[8].split(';')[1].split('=')[1]
        if info[0] not in list(gene_info_dict.keys()):
            gene_info_dict[info[0]] = {}
        if info[2] == 'gene':
            g_id = id
            if re.search(';Name=',info[8]):
                name = re.match('.*;Name=([\w\-\.: ]+);?',info[8]).group(1)
            else:
                name = re.match('.*;makerName=([\w\-\.: ]+);?',info[8]).group(1)
            gene_info_dict[info[0]][id] = {'strand':info[6],'region':str(int(info[3])-1)+'\t'+info[4],'exons':[],'name':name}
        elif info[2] == 'exon':
            gene_info_dict[info[0]][g_id]['exons'].append(str(int(info[3])-1)+'\t'+info[4])
        else:
            continue
    return gene_info_dict

def main():
    parser = argparse.ArgumentParser(description='Linked gene')
    parser.add_argument('-g','--gff',required=True,help='input gff file')
    parser.add_argument('-u','--upstream',default=5000,help='associated gene upstream interval')
    parser.add_argument('-d','--downstream',default=3000,help='associated gene downstream interval')
    parser.add_argument('-i','--input',required=True,help='input bed file')
    parser.add_argument('-o','--output',required=True,help='output prefix')
    args = parser.parse_args()
    g = args.gff
    with open(g,'r') as gff:
        gff_info_dict = parse_gff(gff)
    b = args.input
    o = os.getcwd() + '/' + args.output + str('.bed')
    upstream = args.upstream
    downstream = args.downstream
    with open(b,'r') as bed:
        with open(o,'w') as out:
            bed_lines = bed.readlines()
            for bed_line in bed_lines:
                in_intergenic = 0
                in_upstream = 0
                in_downstream = 0
                in_gene = 0
                in_exon = 0
                in_intro = 0
                most_close_gene_id = ''
                most_close_gene_name = ''
                most_close_gene_in = ''
                most_close_distance = 0
                bed_line = bed_line.strip()
                bed_info = bed_line.split()

                for gene in gff_info_dict[bed_info[0]].keys():
                    strand = gff_info_dict[bed_info[0]][gene]['strand']
                    [start,end] = gff_info_dict[bed_info[0]][gene]['region'].split('\t')
                    gene_name = gff_info_dict[bed_info[0]][gene]['name']
                    gene_id = gene
                    #up_gene_id = []
                    #down_gene_id = []
                    if strand == '+':
                        if int(bed_info[7]) >= int(start) - int(upstream) and  int(bed_info[7]) < int(start):
                            in_upstream = 1
                            distance = int(start) - int(bed_info[7])
                            if most_close_distance == 0 :
                                most_close_distance = distance
                                most_close_gene_id = gene_id
                                most_close_gene_name = gene_name
                                most_close_gene_in = 'up'
                            elif distance < most_close_distance:
                                most_close_distance = distance
                                most_close_gene_id = gene_id
                                most_close_gene_name = gene_name
                                most_close_gene_in = 'up'
                        elif int(bed_info[7]) > int(end) and  int(bed_info[7]) <= int(end) + int(downstream):
                            in_downstream = 1
                            distance = int(bed_info[7]) - int(end)
                            if most_close_distance == 0:
                                most_close_distance = distance
                                most_close_gene_id = gene_id
                                most_close_gene_name = gene_name
                                most_close_gene_in = 'down'
                            elif distance < most_close_distance:
                                most_close_distance = distance
                                most_close_gene_id = gene_id
                                most_close_gene_name = gene_name
                                most_close_gene_in = 'down'
                        elif int(bed_info[7]) <= int(end) and int(bed_info[7]) >= int(start):
                            in_gene = 1
                            for exon  in gff_info_dict[bed_info[0]][gene]['exons']:
                                [exon_start,exon_end] = exon.split('\t')
                                if int(bed_info[7]) <= int(exon_end) and int(bed_info[7]) >= int(exon_start):
                                    in_exon = 1
                                    break
                                else:
                                    continue
                            if in_exon == 0 :
                                in_intro = 1
                                break
                            elif in_exon == 1 :
                                break
                    if strand == '-':
                        if int(bed_info[7]) > int(end) and  int(bed_info[7]) <= int(end) + int(upstream) :
                            in_upstream = 1
                            distance = int(bed_info[7]) - int(end)
                            if most_close_distance == 0:
                                most_close_distance = distance
                                most_close_gene_id = gene_id
                                most_close_gene_name = gene_name
                                most_close_gene_in = 'up'
                            elif distance < most_close_distance:
                                most_close_distance = distance
                                most_close_gene_id = gene_id
                                most_close_gene_name = gene_name
                                most_close_gene_in = 'up'
                        elif int(bed_info[7]) >= int(start) - int(downstream) and  int(bed_info[7]) < int(start) :
                            in_downstream = 1
                            distance = int(start)- int(bed_info[7])
                            if most_close_distance == 0:
                                most_close_distance = distance
                                most_close_gene_id = gene_id
                                most_close_gene_name = gene_name
                                most_close_gene_in = 'down'
                            elif distance < most_close_distance:
                                most_close_distance = distance
                                most_close_gene_id = gene_id
                                most_close_gene_name = gene_name
                                most_close_gene_in = 'down'
                        elif int(bed_info[7]) <= int(end) and int(bed_info[7]) >= int(start):
                            in_gene = 1
                            for exon  in gff_info_dict[bed_info[0]][gene]['exons']:
                                [exon_start,exon_end] = exon.split('\t')
                                if int(bed_info[1]) <= int(exon_end) and int(bed_info[1]) >= int(exon_start):
                                    in_exon = 1
                                    break
                                else:
                                    continue
                            if in_exon == 0 :
                                in_intro = 1
                                break
                            elif in_exon == 1 :
                                break
                if in_gene == 0 and in_upstream == 0 and in_downstream ==0:
                    in_intergenic = 1
                if in_intergenic == 1 :
                    out.write(bed_line+'\tintergenic_region\t'+'None\t'+'None\n' )
                elif in_gene != 1:
                    if in_upstream == 1 and most_close_gene_in == 'up':
                        out.write(bed_line + '\tupstream_gene_variant\t' + most_close_gene_name + '\t' +  most_close_gene_id+'\n')
                    elif in_downstream == 1 and most_close_gene_in == 'down':
                        out.write(bed_line + '\tdownstream_gene_variant\t' + most_close_gene_name + '\t' +  most_close_gene_id + '\n')
                elif in_gene == 1:
                    if in_exon == 1:
                        out.write(bed_line + '\texon_variant\t' + gene_name + '\t' + gene_id + '\n')
                    elif in_intro == 1:
                        out.write(bed_line + '\tintron_variant\t' + gene_name + '\t' + gene_id + '\n')

if __name__ == '__main__':
    main()



```
