# 7. GO analysis



```bash
#obtain protein sequence 

module load bioinfo-tools
module load samtools
module load gffread 
module load SeqKit/2.4.0
 
GFF_longest=/proj/snic2020-2-19/private/cisco/annotation/NBIS_delivery_Coregonus_albula_09_01_2024/gff/gene_build/cisco_longest.gff
REFGENOME=/proj/snic2020-2-19/private/cisco/qiaoling/new_assembly/European_cisco_w_extra_data.ipa1.8.0.purged.primary.fa

cd /proj/snic2020-2-19/private/cisco/qiaoling/Bam_Files/Annotation/Signal_list/0_gene_list/3_kalix

gffread $GFF_longest -g $REFGENOME -y  cisco_protein.fa

#less cisco_protein.fa  
>COALM00000000001 # name by transcriptID
less gwas3.tsv_gff.filtered.list   | awk '{print $1}' | sort |uniq > gwas3.trans_ID.list

seqkit grep -f gwas3.trans_ID.list cisco_protein.fa > gwas3.out_proteins.fa

cp -a gwas3.out_proteins.fa /home/qiade993/dql/Cisco/new_assembly/


```

## Kobas: annotation and enrichment

```bash
 #online website:  http://bioinfo.org/kobas
 #upload gwas3.out_proteins.fa

#gene list information
less gwas3.tsv_gff.filtered.list | sort |uniq > gwas3_trans_gene_id.txt
GFF_longest=/proj/snic2020-2-19/private/cisco/annotation/NBIS_delivery_Coregonus_albula_09_01_2024/gff/gene_build/cisco_longest.gff

while read transid geneid otherid; do  grep -w $geneid $GFF_longest | awk -v transid="$transid" -v geneid="$geneid" '$3=="gene" {print $1"\t"$4"\t"$5"\t"transid"\t"geneid}'; done < gwas3_trans_gene_id.txt > gwas3.gene.pos.txt


##combine
#sort -k1,1 file1.txt > sorted_file1.txt
#sort -k4,4 file2.txt > sorted_file2.txt
join -1 1 -2 4  gwas3.trans_gene.txt gwas3.gene.pos.txt > gwas.go.gene.info.txt

#3. vi 1.txt #save GO id

#less 1.txt |while read i ; do grep $i gwas3.trans_gene.txt; done
less 1.txt |while read i ; do grep $i gwas.go.gene.info.txt |awk '{print $1"\t"$2"\t"$3"\t"$4"\t"$5}'; done

```





