# 4. genome-wide screen differentiation

## 4.1 association analysis

```bash
#------------------step 0 file needed------------------#
This analysis was based on sites from "03_run_initial_analysis_on_whole_dataset"
#Note: The prefixes of these two files need to remain consistent
#1. bam list file： .list
head -n 5 gwas_FegS_vs_FegA.list
FegS_10_ALL_pe_bt2_European_cisco.hifiasm0.16.1_minq20_sorted_overlapclipped_realigned.bam
FegS_11_ALL_pe_bt2_European_cisco.hifiasm0.16.1_minq20_sorted_overlapclipped_realigned.bam
FegS_12_ALL_pe_bt2_European_cisco.hifiasm0.16.1_minq20_sorted_overlapclipped_realigned.bam
FegS_13_ALL_pe_bt2_European_cisco.hifiasm0.16.1_minq20_sorted_overlapclipped_realigned.bam
FegS_14_ALL_pe_bt2_European_cisco.hifiasm0.16.1_minq20_sorted_overlapclipped_realigned.bam

#2. case control file： .ybin
for i in {1..30}; do echo 0; done > gwas_FegS_vs_FegA.ybin
for i in {1..30}; do echo 1; done >> gwas_FegS_vs_FegA.ybin

#------------------step 1 run angsd------------------#
#!/bin/bash -l

#SBATCH -A uppmax2025-2-114
#SBATCH -p node
#SBATCH -n 1
#SBATCH -C mem256GB
#SBATCH -t 10-00:00:00
#SBATCH --array=1-4
#SBATCH -J asso
#SBATCH -e asso_%A_%a.err
#SBATCH -o asso_%A_%a.out
#SBATCH --mail-type=all
#SBATCH --mail-user=qiaoling.deng@outlook.com

#Load modules
module load bioinfo-tools
module load ANGSD/0.933
#module load python/2.7.15


######################################################################################
# CONDUCT GWAS USING ANGSD

# Qiaoling Deng,  2025,5,18
######################################################################################

#SET UP GWAS list

#Specify which bam to use
GWAS_LIST=/proj/snic2020-2-19/private/cisco/qiaoling/Bam_Files/ANGSD_analysis/Beagle_file/2_MAF_005/2_gwas/gwas_list.txt

GWAS_NAMES=$(cat $GWAS_LIST | sed -n ${SLURM_ARRAY_TASK_ID}p)

GWAS_DIR=/proj/snic2020-2-19/private/cisco/qiaoling/Bam_Files/ANGSD_analysis/Beagle_file/2_MAF_005/2_gwas

#Specify which bam to use
BAM_TARGET=${GWAS_NAMES/.list/}
OUTPUT=$(basename $BAM_TARGET)


#Sites
SITES=/proj/snic2020-2-19/private/cisco/qiaoling/Bam_Files/ANGSD_analysis/Beagle_file/2_MAF_005/ANGSD_Cisco_aassembly2_MAF_005_sites.txt

#Specify the number of individuals
x=`cat $GWAS_NAMES | wc -l`


#STEP 1: Define paths to Refference genome
REFGENOME=/proj/snic2020-2-19/private/cisco/qiaoling/new_assembly/European_cisco_w_extra_data.ipa1.8.0.purged.primary.fa
REF_INDEXED=/proj/snic2020-2-19/private/cisco/qiaoling/new_assembly/European_cisco_w_extra_data.ipa1.8.0.purged.primary.fa.fai


#Out directory
OUT_DIR=/proj/snic2020-2-19/private/cisco/qiaoling/Bam_Files/ANGSD_analysis/Beagle_file/2_MAF_005/2_gwas

#Go to the bam files directory
BASEDIR=/proj/snic2020-2-19/private/cisco/qiaoling/Bam_Files/Final_Merged_Bam
cd $BASEDIR

#STEP 2: The association can then be performed on the genotype probabilities using the score statistics
angsd -yBin $GWAS_DIR/$BAM_TARGET.ybin -doAsso 1 -out $OUT_DIR/$OUTPUT  \
-ref $REFGENOME -fai $REF_INDEXED \
-doMajorMinor 4 -doMaf 1 -bam $GWAS_DIR/$GWAS_NAMES -nInd x \
-gl 2 \
-sites $SITES \
-nThreads 8

#------------------output file------------------#
#gwas_cisco_all_bam.lrt0.gz
```



## 4.2 genome-wide screen visualization

```R
#----------------------------------------------------------------------------------------------------------------------------------------
#step1: p VALUE
#----------------------------------------------------------------------------------------------------------------------------------------
setwd("D:/1_Cisco_remapping/7_remapping/2_ANGSD/2_GWAS")

library(tidyverse)
library(qqman)
library(ggplot2)
library(ggpubr)

########################################################################
Cisco_cons2_gwas_FegS_vs_FegA_Sto <- read.delim( "gwas_FegS_vs_FegA.lrt0.gz",header = T)

head(Cisco_cons2_gwas_FegS_vs_FegA_Sto)
colnames(Cisco_cons2_gwas_FegS_vs_FegA_Sto) <- c("Chromosome", "Position", "Major", 
                                                 "Minor", "Frequency" , "LRT")
###############
ref_fai <- read.table("European_cisco_w_extra_data.ipa1.8.0.purged.primary.fa.fai",header=F, sep = "\t")

################
ref_fai_sum_of_second_column <- sum(ref_fai[, 2])


head(ref_fai)
colnames(ref_fai) <- c("contig", "size", "location", 
                       "basesPerLine", "bytesPerLine")

#Sort and remove the unplaced contigs from the assosiation file
ref_fai_contigs <-data.frame(ref_fai[,1:2])
colnames(ref_fai_contigs) <- c("Chromosome", "size")
head(ref_fai_contigs)
ref_fai_contigs <- ref_fai_contigs %>% arrange(desc(size))

ref_fai_contigs$offset <- 
  c(0,cumsum(as.numeric(ref_fai_contigs$size)) ) [-dim(ref_fai_contigs)[1]]

head(ref_fai_contigs)


#Combine with the ref_fai_contigs and offsets

Cisco_cons2_gwas_FegS_vs_FegA_Sto_offset <- Cisco_cons2_gwas_FegS_vs_FegA_Sto %>%
  left_join(ref_fai_contigs, "Chromosome")

Cisco_cons2_gwas_FegS_vs_FegA_Sto_offset <- Cisco_cons2_gwas_FegS_vs_FegA_Sto_offset %>% arrange(desc(size))

Cisco_cons2_gwas_FegS_vs_FegA_Sto_offset_filter <- 
  Cisco_cons2_gwas_FegS_vs_FegA_Sto_offset[-c(
    which(Cisco_cons2_gwas_FegS_vs_FegA_Sto_offset$LRT == -999),
    which(Cisco_cons2_gwas_FegS_vs_FegA_Sto_offset$LRT <= 0)),]

#Add cumsum to the files 
Cisco_cons2_gwas_FegS_vs_FegA_Sto_offset_filter$cum_sum <-Cisco_cons2_gwas_FegS_vs_FegA_Sto_offset_filter$Position + Cisco_cons2_gwas_FegS_vs_FegA_Sto_offset_filter$offset

#Convert our LRT values to p-values. use the command [dchisq] to get pvalues.
Cisco_cons2_gwas_FegS_vs_FegA_Sto_offset_filter$pvalue <- 
  pchisq(Cisco_cons2_gwas_FegS_vs_FegA_Sto_offset_filter$LRT, df = 1, lower.tail= F)
######Calculate the bonferroni threshold
head(Cisco_cons2_gwas_FegS_vs_FegA_Sto_offset_filter)
-log10(0.9)
-log10(0.8)
-log10(0.7)
format (0.05/length(Cisco_cons2_gwas_FegS_vs_FegA_Sto_offset_filter$Position), scientific = F)

gwas2_FegS_vs_FegA_Sto <- Cisco_cons2_gwas_FegS_vs_FegA_Sto_offset_filter
save(gwas2_FegS_vs_FegA_Sto, file = "gwas6_FegS_vs_FegA.Rdata")


#----------------------------------------------------------------------------------------------------------------------------------------
#step 2:manhantton plot: change scaffold names
#----------------------------------------------------------------------------------------------------------------------------------------
#setwd("D:/1_Cisco_remapping/7_remapping/2_ANGSD/2_GWAS")
#load("gwas6_FegS_vs_FegA.Rdata")
library(tidyverse)
library(qqman)
library(ggplot2)
library(ggpubr)
#-----------------------
library(dplyr)
library(CMplot)
library(tibble)

GWAS_FegS_vs_FegA_Sto <- gwas2_FegS_vs_FegA_Sto
GWAS_FegS_vs_FegA_Sto$newChromosome <- GWAS_FegS_vs_FegA_Sto$Chromosome
GWAS_FegS_vs_FegA_Sto$newChromosome <- sub("^RL_", "", GWAS_FegS_vs_FegA_Sto$newChromosome)

#set scaffold as S
GWAS_FegS_vs_FegA_Sto <- GWAS_FegS_vs_FegA_Sto_ %>%
  mutate(newChromosome = ifelse(grepl("scaffold|unloc|bla", newChromosome),"S", newChromosome))

#GWAS_FegS_vs_FegA_Sto_with_mt$newChromosome[is.na(GWAS_FegS_vs_FegA_Sto_with_mt$newChromosome)] <- "NA"
##check
unique(GWAS_FegS_vs_FegA_Sto$newChromosome)
head(GWAS_FegS_vs_FegA_Sto)
### CMplot format: "SNP" "Chromosome" "Position" "P.value"
GWAS_FegS_vs_FegA_Sto <- GWAS_FegS_vs_FegA_Sto[,c("newChromosome","Position","pvalue")]
# 
GWAS_FegS_vs_FegA_Sto <- rownames_to_column(GWAS_FegS_vs_FegA_Sto, var = "RowID")
head(GWAS_FegS_vs_FegA_Sto)
##save
save(GWAS_FegS_vs_FegA_Sto,file="gwas6_FegS_vs_FegA.Rdata")

#----------------------------------------------------------------------------------------------------------------------------------------
#step 3:visualiztion  --- Linux
#----------------------------------------------------------------------------------------------------------------------------------------
# vi GWAS6.r

my_colnames <- c("CHR","BP","winend","snpnumber","count","P")
#P=depth
cisco_depth_all_chr <- read.table(file="all_chromosomes_sorted_depth_del0_del_chrS_chrnumber_for_plot.txt",header=F,col.names=my_colnames)
cisco_depth_all_chr$CHR <- as.numeric(cisco_depth_all_chr$CHR)


library(CMplot)
#gwas2_cutoff_1 <- quantile(cisco_depth_all_chr$pvalue, 0.00001)
#cutoff_2 <- quantile(cisco_depth_all_chr$depth, 0.2)

  # run CMplot
 gwas2<-CMplot(cisco_depth_all_chr, plot.type="m",  col=c("grey", "#d95f02"),
         LOG10=FALSE, 
         chr.den.col=NULL, cex = 1, cex.axis = 1.5, lwd.axis = 2,
         mar = c(3, 6, 1, 1), cex.lab=2,
         file="tiff", memo="depth_manhatton", dpi=300,
        file.output=TRUE,
        verbose=TRUE, width=24, height=4)
#----------------------------
module load bioinfo-tools
module load R/4.1.1
module load R_packages/4.1.1

Rscript 5_manhatton_plot.R


```

